datacube/utils/cog.py

```diff
@@ -9,6 +9,7 @@
 import numpy as np
 import xarray as xr
 import dask
+from dask.base import is_dask_collection
```

Comment:
I think it's fine to import from the top level module, or to call from the top level module. The [examples in the dask API docs](https://docs.dask.org/en/stable/api.html#dask.is_dask_collection) just `import dask` then `dask.is_dask_collection` or one of the other general dask functions.

Reply:
I agree that importing from a higher level module is generally fine (and usually preferred). Importing from the lower level module risks tying our codebase to the current internals of an upstream package which may  be subject to change.